src/Select.js

```diff
@@ -589,10 +589,37 @@ var Select = React.createClass({
 				}
 			}
 			this.setState(newState);
-			if (callback) {
-				callback.call(this, newState);
-			}
+			if (callback) callback.call(this, newState);
 		});
+
+		if (asyncOpts && typeof asyncOpts.then === 'function') {
+			this.props.asyncOptions(input).then((data) => {
```

Comment:
You're re-calling `this.props.asyncOptions(input)` rather than just using `asyncOpts.then`